### ISSUE TITLE
Matrix->IRC->Matrix nick self-feedback (aka "foo[m] (IRC)")

### ISSUE THREAD

**Issue by hydrargyrum** [matrix-org/matrix-appservice-irc#871]
I'm experiencing this issue on a custom Matrix instance bridged to an IRC server.

When a matrix user named "foo" joins a IRC-mapped room:
1. "foo" joins the matrix room, which is also expected (matrix side)
2. "foo[m]" joins the IRC channel, which is normal (matrix->IRC)
3. but "foo[m] (IRC)" also joins the matrix room, which is abnormal (matrix->IRC->matrix)

Worse, people can open conversations with "foo[m] (IRC)" or invite them to conversations but "foo" will not receive the messages and no indication that messages were not received. Also, in the "invite people to room" dialog, the fake user "foo[m] (IRC)" is suggested in addition to "foo", so people are confused.

This also happens on IRC-mapped rooms if "foo" changes nick to "bar": in the matrix room, there will be "bar[m] (IRC)" in addition to "foo[m] (IRC)"! 2 completely useless fake users adding extra confusion.

**Comment by Half-Shot** [comment]
Sounds similar to https://github.com/matrix-org/matrix-appservice-irc/issues/735

**Comment by hydrargyrum** [comment]
@Half-Shot that's not the same problem, not only there's the old and new nicks, but there's also the "[m] (IRC)" nick, and that happens even if the user doesn't change nick

**Comment by Half-Shot** [comment]
> @Half-Shot that's not the same problem, not only there's the old and new nicks, but there's also the "[m] (IRC)" nick, and that happens even if the user doesn't change nick

Yes, hence why I said "similar". It's probably another symptom of the problem where virtual IRC users are not correctly detected in all cases.

**Comment by rbignon** [comment]
Similarly, when a matrix user changes its nickname, we see a new IRC guy joining:

![DeepinScreenshot_select-area_20200210124022](https://user-images.githubusercontent.com/20893520/74146982-92842200-4c02-11ea-9570-b80438ad6667.png)
